File: semifront/src/service/tagService.ts
import api from "../api/axios";
import { Tag } from "../types/type";

export const tagService = {
  // 모든 태그 목록 조회
  getAllTags: () => api.get<Tag[]>("/recipe/tags"),

  // 새 태그 생성 (관리자용)
  createTag: (tag: Tag) => api.post("/tags", tag),

  // 태그 삭제 (관리자용)
  deleteTag: (tagId: number) => api.delete(`/tags/${tagId}`),

  // 특정 레시피에 태그 할당/수정
  assignTagsToRecipe: (recipeId: string, tagIds: number[]) =>
    api.post(`/tags/recipe/${recipeId}`, tagIds),

  // 특정 레시피의 태그 목록 조회
  getRecipeTags: (recipeId: string) =>
    api.get<Tag[]>(`/tags/recipe/${recipeId}`),
};

File: semiproject/src/main/java/org/cloud/service/TagService.java
package org.cloud.service;

import java.util.List;

import org.cloud.dto.Tag;
import org.cloud.mapper.TagMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class TagService {

    @Autowired
    private TagMapper tagMapper;

    public boolean createTag(Tag tag) { return tagMapper.insertTag(tag) > 0; }
    public boolean modifyTag(Tag tag) { return tagMapper.updateTag(tag) > 0; }
    public boolean removeTag(int tagId) { return tagMapper.deleteTag(tagId) > 0; }
    public List<Tag> findAllTags() { return tagMapper.getAllTags(); }

    @Transactional
    public void addTagsToRecipe(String recipeId, List<Integer> tagIds) {
        tagMapper.deleteRecipeTags(recipeId);

        if (tagIds != null && !tagIds.isEmpty()) {
            for (int tagId : tagIds) {
                tagMapper.insertRecipeTag(recipeId, tagId);
            }
        }
    }

    public List<Tag> getRecipeTags(String recipeId) {
        return tagMapper.getTagsByRecipeId(recipeId);
    }
}